### PR TITLE
Enables access to React native developer menu in Eigen

### PR DESCRIPTION
**Pull request by l2succes**
Related: https://github.com/artsy/emission/pull/1345
Also registers new fair screens routes
- `/fair/:id/artists`
- `/fair/:id/artworks`
- `/fair/:id/exhibitors`

~TODO: update Emission version before merging~

### CODE REVIEW

File: Artsy.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -5293,8 +5293,8 @@
 				"${PODS_ROOT}/Artsy+UIFonts/Pod/Assets/Unica77LL-Regular.otf",
 				"${PODS_ROOT}/Artsy+UILabels/Pod/Assets/Chevron_Black.png",
 				"${PODS_ROOT}/Artsy+UILabels/Pod/Assets/Chevron_Black@2x.png",
-				"${PODS_ROOT}/Emission/Pod/Assets/Emission.js",
-				"${PODS_ROOT}/Emission/Pod/Assets/assets",
+				"${PODS_ROOT}/../../emission/Pod/Assets/Emission.js",
```
Comment (l2succes) [inline Artsy.xcodeproj/project.pbxproj]: To update after new Emission version is published

File: Artsy/App/ARAppDelegate+Emission.m [hunk 2]
```diff
@@ -123,6 +124,10 @@ - (void)setupSharedEmissionWithPackagerURL:(NSURL *)packagerURL;
                                                                                   options:options];
 
     AREmission *emission = [[AREmission alloc] initWithConfiguration:config packagerURL:packagerURL];
+    
+    // Disable default React Native dev menu shake motion handler
+    RCTSwapInstanceMethods([UIWindow class], @selector(RCT_motionEnded:withEvent:), @selector(motionEnded:withEvent:));
```
Comment (orta) [inline Artsy/App/ARAppDelegate+Emission.m]: Nice work on that comment :D

File: Artsy/App/ARAppDelegate.m [hunk 3]
```diff
@@ -49,6 +49,7 @@
 #import <DHCShakeNotifier/UIWindow+DHCShakeRecognizer.h>
 #import <VCRURLConnection/VCR.h>
 #import <ObjectiveSugar/ObjectiveSugar.h>
+#import <React/RCTDevSettings.h>
```
Comment (orta) [inline Artsy/App/ARAppDelegate.m]: This might not be needed?

File: Podfile [hunk 4]
```diff
@@ -186,7 +194,11 @@ post_install do |installer|
 
   react = installer.pods_project.targets.find { |target| target.name == 'React' }
   react.build_configurations.each do |config|
-    config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] = '$(inherited) RCT_DEV=0'
+    if config.name == 'Debug'
+      config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] = '$(inherited) RCT_DEV=1'
+    else
+      config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] = '$(inherited) RCT_DEV=0'
+    end
```
Comment (orta) [inline Podfile]: Someone will eventually do refactor this, so:

```
allow_react_native_debugging = config.name == 'Debug' ? '1' : '0'
config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] = '$(inherited) RCT_DEV=' + allow_react_native_debugging
```

it also explains what the point is

**Comment by orta** [comment]
I'm semi interested to see how CI handles the fact that we have different subspecs in dev vs CI - it's likely it will kill `cocoapods-check` (but maybe we can make it ignore a particular pod in the future via config?)

File: Artsy/View_Controllers/Admin/ARAdminSettingsViewController.m [hunk 5]
```diff
@@ -180,7 +179,8 @@ - (ARCellData *)generateRemoteDebug
 - (ARCellData *)generateShowReactNativeDevMenu
 {
     return [self tappableCellDataWithTitle:@"Show React Native Dev Menu" selection:^{
-        [[NSNotificationCenter defaultCenter] postNotificationName:RCTShowDevMenuNotification object:nil];
+        // It'd be nice to use the constant here, but it won't compile on CI
```
Comment (ashfurrow) [inline Artsy/View_Controllers/Admin/ARAdminSettingsViewController.m]: Nice 👍

**Pull request merged**